TPI/app/src/main/java/biz/stratadigm/tpi/presenter/VenuesPresenter.java

```diff
@@ -0,0 +1,66 @@
+package biz.stratadigm.tpi.presenter;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+
+import biz.stratadigm.tpi.interactor.VenuesInteractor;
+import biz.stratadigm.tpi.manager.AppSchedulers;
+import biz.stratadigm.tpi.ui.view.VenuesView;
+import biz.stratadigm.tpi.entity.dto.VenueDTO;
+import biz.stratadigm.tpi.entity.vo.VenueVO;
+
+public class VenuesPresenter extends BasePresenter<VenuesView> {
+    
+    private static final String TAG = "TPI";
+    private final VenuesInteractor venuesInteractor;
+
+    public VenuesPresenter(Context applicationContext,
+                              AppSchedulers appSchedulers,
+                              VenuesInteractor venuesInteractor) {
+        super(applicationContext, appSchedulers);
+        Log.v(TAG, "VenuesPresenter: ctor");
+        this.venuesInteractor = venuesInteractor;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedState) {
+        super.onCreate(savedState);
+        Log.v(TAG, "VenuesPresenter: onCreate");
+
+        executeRequest(venuesInteractor.getVenues(0), new VenuesSubscriber());
+
+    }
+
+    private class VenuesSubscriber extends SimpleSubscriber<ArrayList<VenueDTO>> {
+
+        @Override
+        public void onNext(ArrayList<VenueDTO> venues) {
+            Log.v(TAG, "VenueSubscriber.onNext");
+            ArrayList<VenueVO> vvenues = new ArrayList();
+            Log.v(TAG, ""+venues.size());
+            for (VenueDTO v : venues) {
+                //VenueVO vv = new VenueVO(v.getId(), v.getName(), 0, v.getLoc().getLat(), v.getLoc().getLng());
+                VenueVO vv = new VenueVO(v.getId(), v.getName(), 0, 0f, 0f);
```

Comment:
Please don't put converter logic into presenters. That's why View Objects are made for. The intention of VO objects is to decouple backend entitites from presentation of those entitites in UI. Create a class named VenueConverter that contains a medhod "toVenueVO" which should convert the given VenueDTO to VenueVO. The reason for that: you can use VenueVO across all the application and if API changes somehow, you won't have to change converter logic in several places, instead you'll have to change VenueConverter class only